metals/src/main/scala/scala/meta/internal/metals/BuildTargetInfo.scala

```diff
@@ -0,0 +1,306 @@
+              .sourceJarFile(sourceJarName)
+              .exists(_.toFile.exists())
+            if (hasSource)
+              "         "
+            else
+              "NO SOURCE"
+          }
+        } else " MISSING "
+        val fullName = if (f.toFile.isFile) s" $f" else ""
+        info += s"  $filename$padding $status$fullName"
+      })
+    } else
+      info += "  NONE"
+  }
+
+  private case class TargetDetail(
+      common: CommonTargetDetail,
+      java: Option[JavaTargetDetail],
+      scala: Option[ScalaTargetDetail]
+  ) {
+    def sameJavaScalaClasspaths: Boolean =
+      java.nonEmpty && scala.nonEmpty && java.get.classPath == scala.get.classPath
+    def sameJavaScalaClassesDir: Boolean =
+      java.nonEmpty && scala.nonEmpty && java.get.classesDirectory == scala.get.classesDirectory
+  }
+  private case class CommonTargetDetail(
+      id: BuildTargetIdentifier,
+      displayName: String,
+      baseDirectory: String,
+      tags: List[String],
+      languageIds: List[String],
+      dependencies: List[String],
+      dependentTargets: List[String],
+      capabilities: List[(String, Boolean)],
+      sources: Iterable[AbsolutePath]
+  )
+  private case class JavaTargetDetail(
+      classesDirectory: String,
+      classPath: List[Path],
+      options: List[String]
+  )
+  private case class ScalaTargetDetail(
+      classesDirectory: String,
+      classPath: List[Path],
+      options: List[String],
+      scalaVersion: String,
+      scalaBinaryVersion: String,
+      scalaPlatform: String,
+      jvmVersion: Option[String],
+      jvmHome: Option[String]
+  )
+  private def extractTargetDetail(
+      targetId: BuildTargetIdentifier
+  ): Option[TargetDetail] = {
+    val commonInfo =
+      buildTargets.commonTarget(targetId).map(extractCommonTargetDetail)
+    val scalaInfo =
+      buildTargets.scalaTarget(targetId).map(extractScalaTargetDetail)
+    val javaInfo =
+      buildTargets.javaTarget(targetId).map(extractJavaTargetDetail)
+    commonInfo.map(f => TargetDetail(f, javaInfo, scalaInfo))
+  }
+  private def extractDependencies(target: CommonTarget) = {
+    target.dependencies
+      .map(f =>
+        buildTargets
+          .commonTarget(f)
+          .map(_.displayName)
+          .getOrElse("Unknown target")
+      )
+  }
+  private def extractDependentTargets(target: CommonTarget) = {
+    buildTargets.allCommon
+      .filter(dependentTarget =>
+        dependentTarget.dependencies.contains(target.id)
+      )
+      .map(_.displayName)
+      .toList
+  }
+  private def extractCapabilities(
+      target: CommonTarget
+  ): List[(String, Boolean)] = {
+    var capabilities = List.empty[(String, Boolean)]
+    capabilities ::= ("Debug", target.capabilities.getCanDebug)
+    capabilities ::= ("Test", target.capabilities.getCanTest)
+    capabilities ::= ("Run", target.capabilities.getCanRun)
+    capabilities ::= ("Compile", target.capabilities.getCanCompile)
+    capabilities
```

Comment:
Lets avoid the `var` here. We can just create a list and add these when we create the list like `one :: two :: three :: nil` and just return that. Although again as I stated above, we should avoid a tuple here, and even avoid the extra class if we can.